src/backend/weaviate/weaviate_manager.py

```diff
@@ -64,4 +83,180 @@ def add_weaviate_customer_class(self,customer_guid):
+                    raise ValueError(f"Missing filename in metadata for entry {index}")
+
+                chunk_data = {
+                    "text": entry["text"].strip(),
+                    "customer_guid": customer_guid,
+                    "filename": metadata["filename"],
+                    "chunk_number": metadata["chunk_number"],
+                    "page_numbers": metadata["page_numbers"]
+                }
+
+                # Insert the data object into Weaviate
+                self.client.data_object.create(
+                    data_object=chunk_data,
+                    class_name=class_names,
+                    vector=embedding
+                )
+
+            logger.info(f"Data inserted for {class_names} successfully!")
+        except Exception as e:
+            logger.error(f"Unexpected error inserting data for {customer_guid}: {e}")
+            raise e
+
+    def search_query(self, customer_guid, question, alpha=0.5):
+        try:
+            # Get the query vector for the question
+            query_vector = self.model.encode(question).tolist()
+
+            class_names = self.generate_weaviate_class_name(customer_guid)
+
+            # Perform the query with the provided vector
+            result = self.client.query.get(
+                class_names, ["text", "chunk_number", "page_numbers", "filename", "customer_guid"]
+            ).with_near_vector({
+                "vector": query_vector,
+                "certainty": alpha
+            }).do()
+
+            if not result or "data" not in result or "Get" not in result["data"]:
+                raise ValueError(f"Unexpected search result format: {result}")
+
+            if class_names not in result["data"]["Get"]:
+                raise ValueError(f"No results found for customer: {class_names}")
+
+            for obj in result["data"]["Get"][class_names]:
+                if obj["customer_guid"] != customer_guid:
+                    raise ValueError("Internal server error: Customer GUID mismatch detected!")
+
+            logger.info(f"Search query successful for {customer_guid} with query '{question}'")
+            return result
+
+        except Exception as e:
+            logger.error(f"Unexpected error in search query for {customer_guid}: {e}")
+            raise e
+
+    def delete_objects_by_customer_and_filename(self, customer_guid, filename):
+        try:
+            class_name = self.generate_weaviate_class_name(customer_guid)
+
+            # Define the filter to find objects to delete
+            where_filter = {
+                "operator": "And",
+                "operands": [
+                    {"path": ["customer_guid"], "operator": "Equal", "valueText": customer_guid},
+                    {"path": ["filename"], "operator": "Equal", "valueText": filename}
+                ]
+            }
+            # Fetch the objects to delete
+            objects_to_delete = self.client.query.get(
```

Comment:
I think we had a review comment to delete the objects in the weaviate itself by shortlisting based on a query
Now we had put the earlier version of the code where we are getting objects , and then deleting it one by one

Shift it back to deleting objects in weaviate itself based on the filter. Your earlier versions of the PR had the fix. You can put that fix